src/components/questions/answer.js

```diff
@@ -1,12 +1,87 @@
 import React from "react";
-import formatDate from "../../formatDate"
+import formatDate from "../../formatDate";
+import axios from "axios";
 
 class Answer extends React.Component{
 constructor(props){
   super(props);
-  this.state = {};
+  this.state = {
+    isReported:false
+  };
+  this.helpfulClick = this.helpfulClick.bind(this);
+  this.sendHelpful = this.sendHelpful.bind(this);
+  this.reportedClick = this.reportedClick.bind(this);
+  this.sendReported = this.sendReported.bind(this);
 }
 
+//INITIALIZES LOCAL STORAGE OBJECTS FOR ANSWERS ON MOUNT
+componentDidMount(){
+  localStorage.setItem("helpfulAnswers",JSON.stringify([]));
+  localStorage.setItem("reportedAnswers",JSON.stringify([]));
+}
+
+
+//CHECKS LOCAL STORAGE FOR PREVIOUS ANSWERS MARKED HELPFUL AND IF HELPFUL CLICK
+//CORROSPONDS TO AN ANSWER NOT CONTAINED ON LOCAL STORAGE LIST THEN IT SENDS A PUT REQUEST
+//TO THE API AND RELOADS THE ANSWERS
+
+helpfulClick(event){
+  let helpfulAnswers = localStorage.getItem("helpfulAnswers");
+  helpfulAnswers = JSON.parse(helpfulAnswers);
+  if (helpfulAnswers.indexOf(event.target.id) === -1) {
+    helpfulAnswers.push(event.target.id);
+    localStorage.setItem("helpfulAnswers", JSON.stringify(helpfulAnswers));
+    this.sendHelpful(event.target.id);
+  }
```

Comment:
To fix above issue you could probably check if(helpfulAnswers) and if it's not there, create the array with the id inside

Reply:
i ended up checking on initialization because the actual issue with it was in a conditional render rather than in one of these functions